aten/src/ATen/native/cuda/ReduceOpsKernel.cu

```diff
@@ -169,21 +169,37 @@ void min_values_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+template <typename scalar_t, typename acc_t=scalar_t>
+void argmax_kernel_cuda_impl(TensorIterator& iter) {
+  gpu_reduce_kernel<scalar_t, int64_t>(
+    iter,
+    ArgMaxOps<acc_t>{},
+    thrust::pair<acc_t, int64_t>(at::numeric_limits<acc_t>::lower_bound(), 0));
+};
+
+template <typename scalar_t, typename acc_t=scalar_t>
+void argmin_kernel_cuda_impl(TensorIterator& iter) {
+  gpu_reduce_kernel<scalar_t, int64_t>(
+    iter,
+    ArgMinOps<acc_t>{},
+    thrust::pair<acc_t, int64_t>(at::numeric_limits<acc_t>::upper_bound(), 0));
+};
+
 void argmax_kernel_cuda(TensorIterator& iter) {
+  if (iter.dtype(1) == kHalf) {
+    argmax_kernel_cuda_impl<at::Half, float>(iter);
```

Comment:
worth a comment to describe why are you doing it this way, since max is easily computable for half.

Reply:
done